postr/api_interface.py

```diff
@@ -0,0 +1,40 @@
+# api interface that all api classes must extend
+import abc
+
+
+class ApiInterface(abc.ABC):
+
+    @abc.abstractmethod
```

Comment:
Can you add typing for arguments and return types across all files?  Will be pretty short (20 mins) and important for API design
https://docs.python.org/3/library/typing.html